vsphere/datadog_checks/vsphere/api.py

```diff
@@ -257,13 +257,48 @@ def get_new_events(self, start_time):
         query_filter = vim.event.EventFilterSpec()
         time_filter = vim.event.EventFilterSpec.ByTime(beginTime=start_time)
         query_filter.time = time_filter
-        return event_manager.QueryEvents(query_filter)
+        try:
+            events = event_manager.QueryEvents(query_filter)
+        except SoapAdapter.ParserError as e:
+            self.log.debug("Error parsing bulk events: %s", e)
+
+            if self.config.use_collect_events_fallback:
+                self.log.debug("Start fetch events one by one...")
+                events = self._get_new_events_one_by_one(query_filter)
+            else:
+                raise
+        return events
+
+    def _get_new_events_one_by_one(self, query_filter):
+        # type: (vim.event.EventFilterSpec) -> List[vim.event.Event]
+        """
+        Collecting events one by one and skip those with parsing error.
 
-    @smart_retry
-    def get_latest_event_timestamp(self):
-        # type: () -> dt.datetime
+        The parsing error is triggered by unknown types like `ContentLibrary`.
+        More info:
+            - https://github.com/vmware/pyvmomi/issues/190
+            - https://github.com/vmware/pyvmomi/issues/872
+
+        The event collection fallback is a workaround and can be removed when the upstream issues
+        mentioned above are solved.
+        """
         event_manager = self._conn.content.eventManager
-        return event_manager.latestEvent.createdTime
+        events = []
+        event_collector = event_manager.CreateCollectorForEvents(query_filter)
+        while True:
+            try:
+                collected_events = event_collector.ReadNextEvents(1)  # Read with page_size=1
+            except SoapAdapter.ParserError as e:
+                self.log.debug("Cannot parse event, skipped: %s", e)
+                continue
+            if len(collected_events) == 0:
+                break
+            event = collected_events[0]
+            self.log.debug(
+                "Collect event number:%s, type:%s: msg:%s", event.key, type(event), event.fullFormattedMessage
+            )
+            events.extend(collected_events)
```

Comment:
```suggestion
            events.append(event)
```
?

Reply:
It's equivalent for now, but I think it's safer to keep the current way in case the `page_size` is changed later for some reason.